src/binfhe/lib/binfhecontext.cpp

```diff
@@ -137,7 +259,14 @@ void BinFHEContext::GenerateBinFHEContext(BINFHEPARAMSET set,
   }
 }
 
-LWEPrivateKey BinFHEContext::KeyGen() const {
+LWEPrivateKey BinFHEContext::KeyGen(NativeInteger DiffQ) const {
+  if(DiffQ > m_params->GetLWEParams()->Getq()){
+    auto q = m_params->GetLWEParams()->Getq();
+    this->ChangeQ(DiffQ);
+    auto ret = m_LWEscheme->KeyGen(m_params->GetLWEParams());
```

Comment:
Can we get rid of the variable "res" and simply return "m_LWEscheme->KeyGen(m_params->GetLWEParams());"?

Reply:
It actually needs to do ChangeQ after KeyGen